web/client/components/search/SearchBarUtils.js

```diff
@@ -6,6 +6,22 @@
  * LICENSE file in the root directory of this source tree.
 */
 
+const getTotalMaxResults = (searchOptions, maxResults) => {
+    let array1 = [];
+    const servicesArray = searchOptions.services;
+    for (let i = 0; i < servicesArray.length; i++) {
+        if (servicesArray[i].options && !servicesArray[i].options.maxFeatures) {
+            array1.push(maxResults);
+        } else if (servicesArray[i].options && servicesArray[i].options.maxFeatures) {
+            array1.push(servicesArray[i].options.maxFeatures);
+        }
+    }
+    return array1.reduce(function(a, b) {
+        return a + b;
+    }, 0);
+};
```

Comment:
i think like this is better :)
you also need to change the call to this functions since here we pass directly the services array
```suggestion
    const getTotalMaxResults = (services = [], maxResults = 15) => {
    return services.reduce(function(total, current) {
        return total + (current?.options?.maxFeatures || maxResults)
    }, 0) || maxResults;
};
```

Reply:
can you add unit tests only for this function?
you can test it with no params passed, with only the services, with both etc.

you need to export it in order to test it inside SearchBarUtils-test.js